apps/builder/app/canvas/features/webstudio-component/link.ts

```diff
@@ -0,0 +1,53 @@
+import type { MouseEvent } from "react";
+import { useStore } from "@nanostores/react";
+import { findPageByIdOrPath, Page } from "@webstudio-is/project-build";
+import { pagesStore, useIsPreviewMode } from "~/shared/nano-states";
+import { publish } from "~/shared/pubsub";
+
+declare module "~/shared/pubsub" {
+  export interface PubsubMap {
+    switchPage: { pageId: Page["id"] | "home" };
+  }
+}
+
+const isAbsoluteUrl = (href: string) => {
+  try {
+    new URL(href);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const useHandleLinkClick = () => {
+  const pages = useStore(pagesStore);
+  const [isPreviewMode] = useIsPreviewMode();
+
+  return (event: MouseEvent) => {
+    event.preventDefault();
+
+    if (isPreviewMode === false || pages === undefined) {
```

Comment:
Inside event handlers you can access previewModeStore and pagesStore directly to not trigger rerender when any of them are changed.

Reply:
Btw can be just a function without hook closure